src/qgitrepository.cpp

```diff
@@ -520,7 +518,10 @@ void Repository::fetch(const QString& name, const QString& head, const Signature
         refs.set(QList<QByteArray>() << refspec.toLatin1());
     }
 
-    qGitThrow(git_remote_fetch(remote.data(), refs.count() > 0 ? &refs.data() : NULL, signature.data(), message.isNull() ? NULL : message.toUtf8().constData()));
+    internal::RemoteCallbacks remoteCallbacks(d_ptr.data(), d_ptr->m_remote_credentials.value(name));
+    git_fetch_options opts = GIT_FETCH_OPTIONS_INIT;
+    opts.callbacks = remoteCallbacks.rawCallbacks();
+    qGitThrow(git_remote_fetch(remote.data(), refs.count() > 0 ? &refs.data() : NULL, &opts, message.isNull() ? NULL : message.toUtf8().constData()));
```

Comment:
Hmm, I wonder why there's no `fetch` function in `Remote`...

Well, that's not to be fixed here. I think I'll create an issue for that.